### ISSUE TITLE
Unable to start a node because of a missing status field in receipt

### ISSUE THREAD

**Issue by CosminNechifor** [raiden-network/raiden#2664]
## Problem Definition

My node became unusable and it can no longer start because it crashes with the following stacktrace: 

```
Traceback (most recent call last):
  File "/home/cosmin/.virtualenvs/raiden/bin/raiden", line 11, in <module>
    load_entry_point('raiden', 'console_scripts', 'raiden')()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/__main__.py", line 11, in main
    run(auto_envvar_prefix='RAIDEN')  # pylint: disable=no-value-for-parameter
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 722, in __call__
    return self.main(*args, **kwargs)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 697, in main
    rv = self.invoke(ctx)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 1043, in invoke
    return Command.invoke(self, ctx)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 895, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/core.py", line 535, in invoke
    return callback(*args, **kwargs)
  File "/home/cosmin/.virtualenvs/raiden/lib/python3.6/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 1204, in run
    NodeRunner(kwargs, ctx).run()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 1035, in run
    app = self._run_app()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 1054, in _run_app
    app_ = run_app(**self._options)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/ui/cli.py", line 851, in run_app
    raiden_app.start()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/app.py", line 139, in start
    self.raiden.start()
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/raiden_service.py", line 347, in start
    self.raiden_event_handler.on_raiden_event(self, transaction)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/raiden_event_handler.py", line 93, in on_raiden_event
    self.handle_contract_send_channelsettle(raiden, event)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/raiden_event_handler.py", line 449, in handle_contract_send_channelsettle
    partner_locksroot,
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/network/proxies/payment_channel.py", line 214, in settle
    partner_locksroot=partner_locksroot,
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/network/proxies/token_network.py", line 935, in settle
    receipt_or_none = check_transaction_threw(self.client, transaction_hash)
  File "/home/cosmin/Documents/WORK/GITHUB/raiden/raiden/network/rpc/transactions.py", line 8, in check_transaction_threw
    if 'status' not in receipt:
TypeError: argument of type 'NoneType' is not iterable
```
I want to mention that I am using infura and the tx_hash is: https://ropsten.etherscan.io/tx/0x0ca6dad843243eb1ea0ec9bba58864fc2c20b8a5d749c3585ab9112b9a0349fe
Full log: 
[log.txt](https://github.com/raiden-network/raiden/files/2445902/log.txt)

**Comment by hackaugusto** [comment]
@CosminNechifor what ethereum client are you using and what is the version? also, what is the actual content of the response?

**Comment by palango** [comment]
> If I call for the transaction receipt I get this:

Again, from infura or a different node?

**Comment by hackaugusto** [comment]
The first call to `eth_gettransactionreceipt` returned a `null` value, a manual subsequent returned some data. So perhaps was it a reorg?

**Comment by hackaugusto** [comment]
> But I guess we should add a None check in check_transaction_threw anyways.

Just checking if it's a None is not that useful, the only thing we could do is to throw a different exception which is a bit more information. I guess the fix for this is to add a default number of confirmations block to all the `poll(tx_hash)` (assuming the problem was really a reorg)

**Comment by CosminNechifor** [comment]
Is weird because after I  started the geth node it worked normally. And then I decided to try once Again with infura, and it was no longer breaking.

**Comment by LefterisJP** [comment]
Assuming this is a reorg. If you can find a reproducible way to handle this then we can reopen. Until then I am closing this as it will be handled by: https://github.com/raiden-network/raiden/issues/2711

**Comment by palango** [comment]
I don't think this will be handled by @hackaugusto s PR, this is about requiring a certain number of  confirmations in the `RPCClient.poll()` method.

**Comment by LefterisJP** [comment]
@palango I see. Good point.

**Comment by palango** [comment]
Fixed by #2731
